pennylane/tape/interfaces/autograd.py

```diff
@@ -156,7 +194,9 @@ def vjp(ans, self, params, device):  # pylint: disable=unused-argument
         """
 
         def gradient_product(g):
+            self.set_parameters(self._all_params_unwrapped, trainable_only=False)

```

Comment:
Ah, so we don't need to do the unwrapping in `vjp()` because we know that `_execute()` (which does the unwrapping) will have always been called first?

Reply:
Yes that's right - the way `autograd.grad` works, the forward pass is *always* performed first. I'll add a comment
```suggestion
            # In autograd, the forward pass is always performed prior to the backwards
            # pass, so we do not need to re-unwrap the parameters.
            self.set_parameters(self._all_params_unwrapped, trainable_only=False)
```